mbed_project/_internal/project_data.py

```diff
@@ -21,106 +21,64 @@
-    """
-
-    repo_url: str
-    ref: str
-
-
-@dataclass
-class MbedLibReference:
-    """Metadata associated with an Mbed library.
-
-    An Mbed library is an external dependency of an MbedProgram. The MbedProgram is made aware of the library
-    dependency by the presence of a .lib file in the project tree, which we refer to as a library reference file. The
-    library reference file contains a URI where the dependency's source code can be fetched.
-
-    Attributes:
-        reference_file: Path to the .lib reference file for this library.
-        source_code_path: Path to the source code if it exists in the local project.
-    """
-
-    reference_file: Path
-    source_code_path: Path
-
-    def is_resolved(self) -> bool:
-        """Determines if the source code for this library is present in the source tree."""
-        return self.source_code_path.exists() and self.source_code_path.is_dir()
-
-    def get_git_reference(self) -> GitReference:
-        """Get the source code location from the library reference file.
-
-        Returns:
-            Data structure containing the contents of the library reference file.
-        """
-        raw_ref = self.reference_file.read_text().strip()
-        url, sep, ref = raw_ref.partition("#")
-        return GitReference(repo_url=url, ref=ref)
-
-
-@dataclass
-class MbedProgramData:
-    """Metadata associated with an MbedProgram.
-
-    This object holds paths to the various config files that can be found in an MbedProgram.
+    All MbedPrograms must contain a .mbed file, which defines the program root path. MbedPrograms must also contain an
+    mbed-os.lib reference file, defining Mbed OS as a program dependency. Because the mbed-os.lib reference is required
+    by all Mbed programs the path to the file is stored in this object.
 
     Attributes:
-        config_file: Path to mbed_app.json file. This can be `None` if the program doesn't set any custom config.
+        app_config_file: Path to mbed_app.json file. This can be `None` if the program doesn't set any custom config.
         mbed_file: Path to the .mbed file which defines the program root.
         mbed_os_ref: Library reference file for MbedOS. All programs require this file.
     """
 
-    config_file: Optional[Path]
+    app_config_file: Optional[Path]
     mbed_file: Path
     mbed_os_ref: Path
 
     @classmethod
-    def from_new(cls, root_path: Path) -> "MbedProgramData":
-        """Create MbedProgramData from a new directory.
+    def from_new(cls, root_path: Path) -> "MbedProgramFiles":
+        """Create MbedProgramFiles from a new directory.
 
         A "new directory" in this context means it doesn't already contain an Mbed program.
 
         Args:
             root_path: The directory in which to create the program data files.
 
         Raises:
-            ValueError: MbedProgramData already exists at this path.
+            ValueError: MbedProgramFiles already exists at this path.
```

Comment:
Just to check I understand right, from a user's pov that would be that you are trying to do `mbed-tools new` when there already exists a project with a `.mbed` file in it?

Reply:
Yep, that's right. The user would be doing `mbed-tools init` in a directory with a .mbed and/or mbed-os.lib file present.

Reply:
I might be tempted to tweak this a little and say something like `An Mbed Project already exists at this path` to give the context a bit more.

Reply:
e505b1cc9559ef5e2e01d931d6cdcc5ed7484688